lightning/src/ln/channelmonitor.rs

```diff
@@ -1031,12 +1026,34 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 			their_htlc_base_key: &PublicKey, their_delayed_payment_base_key: &PublicKey,
 			their_to_self_delay: u16, funding_redeemscript: Script, channel_value_satoshis: u64,
 			commitment_transaction_number_obscure_factor: u64,
+			initial_local_commitment_tx: LocalCommitmentTransaction,
 			logger: Arc<Logger>) -> ChannelMonitor<ChanSigner> {
 
 		assert!(commitment_transaction_number_obscure_factor <= (1 << 48));
 		let our_channel_close_key_hash = Hash160::hash(&shutdown_pubkey.serialize());
 		let shutdown_script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0).push_slice(&our_channel_close_key_hash[..]).into_script();
 
+		let mut onchain_tx_handler = OnchainTxHandler::new(destination_script.clone(), keys.clone(), their_to_self_delay, logger.clone());
+
+		let local_tx_sequence = initial_local_commitment_tx.without_valid_witness().input[0].sequence as u64;
+		let local_tx_locktime = initial_local_commitment_tx.without_valid_witness().lock_time as u64;
+		let local_commitment_tx = LocalSignedTx {
+			txid: initial_local_commitment_tx.txid(),
+			revocation_key: initial_local_commitment_tx.local_keys.revocation_key,
+			a_htlc_key: initial_local_commitment_tx.local_keys.a_htlc_key,
+			b_htlc_key: initial_local_commitment_tx.local_keys.b_htlc_key,
+			delayed_payment_key: initial_local_commitment_tx.local_keys.a_delayed_payment_key,
+			per_commitment_point: initial_local_commitment_tx.local_keys.per_commitment_point,
+			feerate_per_kw: initial_local_commitment_tx.feerate_per_kw,
+			htlc_outputs: Vec::new(), // There are never any HTLCs in the initial commitment transactions
+		};
+		// Returning a monitor error before updating tracking points means in case of using
```

Comment:
06fc27d

Was this my comment or yours ? Because if watchtower doesn't reject update, it means they have the latest local valid commitment, not us. But if we reject update, because of a block triggering broadcast, our tip local commitment is a valid one too. It's just the prev one?

I think we are robust against this scenario right now or what failure do you consider ? If we reject update (quorum or single-failure only) revocation secret shouldn't have been released offchain?

Reply:
Definitely not mine. It is in provide_latest_local_commitment_tx_info on master with a blame of e46e183084ed858f41aa304acd78503aea1a96ed.

Reply:
I think your point was that if we update one monitor but another refuses the update, then the first monitor may broadcast a new local state, which the second monitor won't have stored anywhere. I believe this is still the case after this PR.

Reply:
Yes, right my comment is confusing as I read it, what I mean here is by latest local valid is a in fact the prev one from the _other_ watchtower viewpoint. Anyway we aren't robust against this, and more I'm thinking there is also dirty reorg-case where one monitor lockdown itself, broadcast commitment, this one get reorg'ed out... Other monitors have always been on the main chain, what happens ?

Reply:
See https://github.com/rust-bitcoin/rust-lightning/issues/604

Reply:
Yea, lets discuss it on the issue. Its def confusing.